mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,19 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+
+        // check if indices is one-based and in ascending order
```

Comment:
Yeah, that's fair enough. I don't think this is a performance-critical block anyway, but the cleverness is probably a step too far in complexity.

This needs a little rebase, and i think you can still use `require` for the check.
This allows `indices(0) == 0`. Do you mean to initialize `previous = 0`?